src/rank_llm/rerank/reranker.py

```diff
@@ -379,11 +379,13 @@ def extract_kwargs(
     # Check that type of provided kwarg is compatible with the provided default type
     for i, extracted_kwarg in enumerate(extracted_kwargs):
         default_value = keys_and_defaults[i][1]
-        if default_value is not None and not isinstance(
-            extracted_kwarg, type(default_value)
+        if (
+            not isinstance(extracted_kwarg, type(default_value))
+            and extracted_kwarg is not None
+            and default_value is not None
         ):
             raise ValueError(
-                f"Provided kwarg for {keys_and_defaults[i][0]} must be compatible with the argument's default type {type(default_value)}"
+                f"Provided kwarg for {keys_and_defaults[i][0]} must be compatible with the argument's default type {type(default_value)}. Got {extracted_kwarg}"
```

Comment:
IIUC, you fix allows overriding of a default non-None value with an extract_kwarg of None, why do we want to allow this?

Please mention the type too:
"...Got {extracted_kwarg} of type {type(extracted_kwarg}"

Reply:
sorry, yes. I have fixed it